tests/std/tests/P2474R2_views_repeat/test.cpp

```diff
@@ -0,0 +1,331 @@
+    const I second = ranges::next(first);
+
+    assert(*first == *second);
+    assert(&*first == &*second);
+    static_assert(noexcept(*first));
+
+    {
+        I tmp = first;
+        static_assert(is_nothrow_copy_constructible_v<I>);
+
+        I i = move(tmp);
+        static_assert(is_nothrow_move_constructible_v<I>);
+
+        tmp = i;
+        static_assert(is_nothrow_copy_assignable_v<I>);
+
+        i = move(tmp);
+        static_assert(is_nothrow_move_assignable_v<I>);
+    }
+
+    assert(!(first == second));
+    assert(!(second == first));
+    static_assert(noexcept(first == second)); // strengthened
+
+    assert(first != second);
+    assert(second != first);
+    static_assert(noexcept(first != second)); // strengthened
+
+    assert(first < second);
+    assert(!(second < first));
+    static_assert(noexcept(first < second)); // strengthened
+
+    assert(!(first > second));
+    assert(second > first);
+    static_assert(noexcept(first > second)); // strengthened
+
+    assert(first <= second);
+    assert(!(second <= first));
+    static_assert(noexcept(first <= second)); // strengthened
+
+    assert(!(first >= second));
+    assert(second >= first);
+    static_assert(noexcept(first >= second)); // strengthened
+
+    assert(first <=> second < 0);
+    assert(second <=> first > 0);
+    static_assert(noexcept(first <=> second)); // strengthened
+
+    {
+        I i = first;
+
+        assert(&++i == &i);
+        assert(i == second);
+        static_assert(noexcept(++i)); // strengthened
+
+        i = first;
+
+        assert(i++ == first);
+        assert(i == second);
+        static_assert(noexcept(i++)); // strengthened
+
+        assert(&--i == &i);
+        assert(i == first);
+        static_assert(noexcept(--i)); // strengthened
+
+        i = second;
+
+        assert(i-- == second);
+        assert(i == first);
+        static_assert(noexcept(i--)); // strengthened
+
+        assert(i + 1 == second);
+        static_assert(noexcept(i + 1)); // strengthened
+
+        assert(1 + i == second);
+        static_assert(noexcept(1 + i)); // strengthened
+
+        assert(&(i += 1) == &i);
+        assert(i == second);
+        static_assert(noexcept(i += 1)); // strengthened
+        assert(i - 1 == first);
+        static_assert(noexcept(i - 1)); // strengthened
+
+        assert(&(i -= 1) == &i);
+        assert(i == first);
+        static_assert(noexcept(i -= 1)); // strengthened
+
+        assert(second - first == 1);
+        static_assert(noexcept(second - first)); // strengthened
+
+        assert(first[1] == *second);
+        assert(second[-1] == *first);
+        static_assert(noexcept(first[1])); // strengthened
+    }
+
+    const same_as<ranges::sentinel_t<R>> auto last = rng.end();
+    const bool is_empty                            = bound == 0;
+    assert(first == last == is_empty);
+    static_assert(noexcept(first == last)); // strengthened
+    assert(first != last != is_empty);

```

Comment:
Although this is correct, it's visually confusing for humans. I recommend introducing parentheses for clarity (one of the few cases where "unnecessary" parentheses are desirable).